modules/ROOT/pages/understanding-enterprise-integration-patterns-using-mule.adoc

```diff
@@ -0,0 +1,92 @@
+The Enterprise Integration Patterns are accepted solutions to recurring problems within a given context and they provide a framework for designing, building messaging and integration systems as well as a common language for teams to use when architecting solutions.
+
+Mule supports most of the Enterprise Integration Patterns from the http://www.eaipatterns.com/toc.html[book] by Gregor Hohpe and Bobby Woolf.
+
+The fact that Mule implements these patterns greatly reduces the effort required when building integrations. You can design your solution using these well-known patterns and then simply configure and use these same patterns in Mule.
+
+== Mapping Enterprise Integration Patterns into Mule Objects
+
+There is a list of patterns that can be mapped directly with Mule objects. Find the list of Enterprise Integration Patterns mapped to Mule objects below:
+
+=== Integration Styles
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|File Transfer |xref:connectors::file/file-connector.adoc[File Connector]
+|Shared Database |xref:connectors::db/db-connector-index.adoc[DataBase Connector]
+|Remote Procedure Invocation | Mule APIs are meant to work like this (or even doing requests to external APIs) 
+|Messaging |Mule is all about Messaging
+|===
+
+
+=== Messaging Systems
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Message Channel |Mule provides a message channel that connects the message processors in a flow.
+|Pipes and Filters |A Flow implements a pipe and filter architecture
+|Message Router |xref:about-components.adoc#flow-control-routers[Message Routers]
+|Message Translator |xref:transform-component-about.adoc[Message Transformer]
+|Message Endpoint | xref:mule-sdk::sources.adoc[Message Sources] and xref:mule-sdk::operations.adoc[Operations]
+|===
+
+=== Messaging Channels
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Point-to-Point Channel |The default channel within a flow.
+|Message Bus |Mule is a message bus.
+|Guaranteed Delivery | Using xref:reliability-patterns.adoc[Reliability Patterns].
+|===
+
+=== Message Construction
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Event Message | Mule works transmiting events from different Application or Processors.
+|Request Reply | Connectors that allow request-reply wise operations, or using xref:reliability-patterns.adoc[Reliability Patterns].
+|===
+
+=== Message Routing
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Content-Based Router |xref:choice-router-concept.adoc[Choice Router]
+|Message Filter |  xref:connectors::validation/validation-connector.adoc[Validation Module]
+|Dynamic Routing| xref:about-components.adoc#flow-control-routers[Message Routers]
+|Scatter Gather | xref:scatter-gather-concept.adoc[Scatter Gather Router].
+|Splitter | xref:for-each-scope-concept.adoc[Foreach Scope], xref:parallel-foreach-scope.adoc[Parallel Foreach Scope] and xref:batch-processing-concept.adoc[Batch].
```

Comment:
As a comment, remember to remove parallel foreach reference when cherrypicking to v4.1